bin/prepare-release.sh

```diff
@@ -0,0 +1,150 @@
+    exit 1
+fi
+
+echo "Preparing update to v${NEW_VERSION} (${DATE})"
+echo "Workspace root: ${WORKSPACE_ROOT}"
+echo
+
+GIT_STATUS_OUTPUT=$(git status --untracked-files=no --porcelain)
+if [ -z "$ALLOW_DIRTY" ] && [ -n "${GIT_STATUS_OUTPUT}" ]; then
+    lines=$(echo "$GIT_STATUS_OUTPUT" | wc -l | tr -d '[:space:]')
+    echo "error: ${lines} files in the working directory contain changes that were not yet committed into git:"
+    echo
+    echo "${GIT_STATUS_OUTPUT}"
+    echo
+    echo 'To proceed despite this and include the uncommited changes, set the `ALLOW_DIRTY` environment variable.'
+    exit 1
+
+fi
+
+DOIT=y
+if [[ -n "$DRY_RUN" ]]; then
+    echo "Dry-run. Not modifying files."
+    DOIT=n
+fi
+
+### GLEAN-CORE ###
+
+FILE=glean-core/Cargo.toml
+run $SED -i.bak -E \
+    -e "s/^version = \"[0-9a-z.-]+\"/version = \"${NEW_VERSION}\"/" \
+    "${WORKSPACE_ROOT}/${FILE}"
+run rm "${WORKSPACE_ROOT}/${FILE}.bak"
+
+### GLEAN-FFI ###
+
+FILE=glean-core/ffi/Cargo.toml
+run $SED -i.bak -E \
+    -e "s/^version = \"[0-9a-z.-]+\"/version = \"${NEW_VERSION}\"/" \
+    -e "/glean-core/!b;n;n;s/version = \"[0-9a-z.-]+\"/version = \"${NEW_VERSION}\"/" \
+    "${WORKSPACE_ROOT}/${FILE}"
+run rm "${WORKSPACE_ROOT}/${FILE}.bak"
+
+### KOTLIN PACKAGES ###
+
+FILE=.buildconfig.yml
+run $SED -i.bak -E \
+    -e "s/libraryVersion: [0-9A-Z.-]+/libraryVersion: ${NEW_VERSION}/" \
+    "${WORKSPACE_ROOT}/${FILE}"
+run rm "${WORKSPACE_ROOT}/${FILE}.bak"
+
+### GLEAN GRADLE PLUGIN ###
+
+FILE=gradle-plugin/src/main/groovy/mozilla/telemetry/glean-gradle-plugin/GleanGradlePlugin.groovy
+run $SED -i.bak -E \
+    -e "s/project.ext.glean_version = \"[0-9A-Z.-]+\"/project.ext.glean_version = \"${NEW_VERSION}\"/" \
+    "${WORKSPACE_ROOT}/${FILE}"
+run rm "${WORKSPACE_ROOT}/${FILE}.bak"
+
+### CHANGELOG ###
+
+FILE=CHANGELOG.md
+run $SED -i.bak -E \
+    -e "s/# Unreleased changes/# v${NEW_VERSION} (${DATE})/" \
+    -e "s/\.\.\.master/...${NEW_VERSION}/" \
+    "${WORKSPACE_ROOT}/${FILE}"
+run rm "${WORKSPACE_ROOT}/${FILE}.bak"
+
+if [ "$DOIT" = y ]; then
+    CHANGELOG=$(cat "${WORKSPACE_ROOT}/${FILE}")
+    cat > "${WORKSPACE_ROOT}/${FILE}" <<EOL
+# Unreleased changes
+
+[Full changelog](https://github.com/mozilla/glean/compare/v${NEW_VERSION}...master)
+
+${CHANGELOG}
+EOL
+fi
+
+echo "Everything prepared for v${NEW_VERSION}"
+echo
+echo "Changed files:"
+git status --untracked-files=no --porcelain || true
+echo
+echo "Create release commit v${NEW_VERSION} now? [y/N]"
+read -r RESP
+echo
+if [ "$RESP" != "y" ] && [ "$RESP" != "Y" ]; then
+    echo "No new commit. No new tag. Proceed manually."
+    exit 0
+fi
+
+run git add --update "${WORKSPACE_ROOT}"
+run git commit --message "Release version ${NEW_VERSION}"
+
+echo
+echo "Create tag v${NEW_VERSION} now? [y/N]"
+read -r RESP
+echo
+if [ "$RESP" != "y" ] && [ "$RESP" != "Y" ]; then
+    echo "No new tag. Proceed manually."
+    exit 0
+fi
+
+run git tag "v${NEW_VERSION}"
```

Comment:
If you add a tag here, and then follow the instructions below to make a pull request, merge it, and then tag using the Github UI, this tag won't really have any effect, right, since it's on the user fork (and will probably just cause confusion because there will be a tag in the user fork pointing to a different commit than that same tag on master).

Generally, I prefer to have scripts automate as much as possible, so it would be nice to just remove the manual github UI step described below to tag the release -- but I don't know how that would work with branch protection.

Reply:
Oh, and I guess this tag won't have the changelog in the commit message like it does using the instructions below.  (It could with some serious sed-fu, probably...)

Reply:
Ah, damn, you are right. Let's not create the tag here then for now.